app/database_etl/test_evaluation_handler/test_eval_link_handler.py

```diff
@@ -0,0 +1,100 @@
+        manufacturers = ['Zhejiang Orient Gene (Biotech Co LTD China)']
+
+    temp = pd.DataFrame()
+    # Get all rows matching any name in manufacturer list
+    for manufacturer in manufacturers:
+        if manufacturer in manufacturer_mapping: manufacturer = manufacturer_mapping[manufacturer]
+        manu = matches[matches['Manufacturer'].str.contains(f"(?i){manufacturer}")]
+        temp = temp.append(manu)
+
+    matches = temp
+    # Check that targets are exact match
+    for i, ev in matches.iterrows():
+        if not set(ev['Target']) == set(target_isotypes):
+            matches.drop(i)
+
+    if matches.empty: return [None]*2
+    if matches.shape[0] == 1: return [matches.iloc[0]['unique_id'], matches.iloc[0]['DOI']]
+
+    # CRITERIA B – sample specimen type
+    specimen_type = row['specimen_type']
+    new_matches = matches[matches['IndexSampleType'] == specimen_type]
+
+    if not new_matches.empty:
+        matches = new_matches
+        if matches.shape[0] == 1: return [matches.iloc[0]['unique_id'], matches.iloc[0]['DOI']]
+
+    # CRITERIA C – prioritized reference specimen type
+    rankings = {'Lower respiratory specimen': 1, 'Sputum': 3, 'Bronchial aspirate': 3,
+                'Respiratory specimen': 4, 'Upper respiratory specimen': 4, 'Nasal mid-turbinate swabs': 5,
+                'Nasopharyngeal swabs': 7, 'Swab': 7, 'Serum': 8, 'Pharyngeal swabs': 10, 'Plasma': 11, 'Oropharyngeal swabs': 12,
+                'Whole blood': 13, 'Urine': 14, 'Not available': 20, 'not reported': 20, 'Not reported': 20}
+
+    min_rank, top_matches = 100, pd.DataFrame()
+    for i, ev in matches.iterrows():
+        if pd.isna(ev['ReferenceSampleType']):
+            if top_matches.empty or min_rank == 20:
+                min_rank = 20
+                top_matches = top_matches.append(ev)
+                continue
+        reference_specimen_types = ev['ReferenceSampleType'].split('; ')
+        for ref in reference_specimen_types:
+            rank = rankings[ref]
+            if rank > min_rank: continue
+            if rank == min_rank: top_matches = top_matches.append(ev)  # Same rank, so append
+            else:  # rank < min_rank
+                min_rank, top_matches = rank, ev.to_frame().T
+
+    if not top_matches.empty:
+        matches = top_matches
+        if matches.shape[0] == 1: return [matches.iloc[0]['unique_id'], matches.iloc[0]['DOI']]
+
+    # CRITERIA D – largest sample size
+    largest_sample_size, top_match = 0, pd.DataFrame()
+    for i, ev in matches.iterrows():
+        sample_size = ev['SampleSize']
+        if sample_size > largest_sample_size:
+            largest_sample_size, top_match = sample_size, ev.to_frame().T
+
+    return [top_match['unique_id'], top_match['DOI']]
+
+
+def apply_test_eval_links(df):
+    df['ind_eval_uid_doi'] = df.apply(lambda row: get_eval_UID_and_DOI(row), axis=1)
+    df[['test_linked_uid', 'ind_eval_link']] = pd.DataFrame(df['ind_eval_uid_doi'].to_list(), index=df.index)
```

Comment:
@abeljoseph -- you link the test in, but you don't actually pull the corresponding values through here. 

The point of the test link is it lets us pull the independent evaluation values for sensitivity and specificity (in particular, `ind_se`, `ind_se_n`, `ind_sp`, `ind_sp_n`) from the FindDx CSV through to each estimate. We've been doing that with a lookup field in Airtable, but now we should just be able to do a direct lookup from the CSV, right? Once we have done this, we should add these to the dataframe and database, and overwrite the values for those fields that are coming from the Airtable. 

@atatmaja, could you confirm that no other fields are needed for test adjustment here?

Reply:
I think we also have a column called 'test_validation', not sure if it's populated by test linking @rahularoradfs 

Some context on how it's being used in the test adjustment script:

https://github.com/serotracker/iit-backend/pull/181/files#diff-22acfea80e41338a5fa26a4264e9cce8c2d39121f24e1605d699f56f8c643a43R179